src/NServiceBus.Core/Unicast/Transport/Config/TransportExtensions.cs

```diff
@@ -28,15 +29,37 @@ public new TransportExtensions<T> ConnectionString(string connectionString)
             return this;
         }
 
+#if NET452
         /// <summary>
         /// Configures the transport to use the connection string with the given name.
         /// </summary>
+        [ObsoleteEx(
+        Message = "Directly using named connection strings is discouraged. Instead, load the connection string in your code and pass the value to TransportExtensions<T>.ConnectionString(connectionString)",
```

Comment:
how about:
~~Directly~~ Using named connection strings is discouraged. Instead, load the connection string in your code and pass the value to EndpointConfiguration.UseTransport<T>().ConnectionString(connectionString).